### PR TITLE
Update ecflow from authoritative spack repository, add py-scipy to jedi-tools-env

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

Two weeks ago, I sent our `ecflow` updates (and a bug fix in the `python` package for `ecflow`) to the upstream spack develop branch. The code changes got merged but then improved (see https://github.com/spack/spack/pull/34043#event-7880586249). This PR is bringing back the changes from the spack develop PR - a direct cherry-pick of the original commits. The changes in `boost`, `ecflow`, `python` are all from that PR.

Additional changes (separate commits):
- add `py-scipy` to `jedi-tools-env`
- update `ewmwf-atlas` and `fckit` to use `args` instead of `res` in `def cmake_args`, in order to be consistent with other cmake packages (and in particular with `eckit`)

## Testing

See https://github.com/NOAA-EMC/spack-stack/pull/409

### CODE REVIEW

File: var/spack/repos/builtin/packages/ecflow/package.py [hunk 1]
```diff
@@ -60,19 +60,27 @@ class Ecflow(CMakePackage):
 
     depends_on("openssl@1:", when="@5:")
     depends_on("qt@5:", when="+ui")
-    depends_on("cmake@2.12.11:", type="build")
+    # Requirement to use the Python3_EXECUTABLE variable
```
Comment (AlexanderRichert-NOAA) [inline var/spack/repos/builtin/packages/ecflow/package.py:63]: Looks like the comment is  out of place?

Comment (climbfuji) [reply var/spack/repos/builtin/packages/ecflow/package.py:63]: No, cmake@3.16 provides -DPython3_EXECUTABLE

**Review by AlexanderRichert-NOAA — DISMISSED** [review]
Just the out of place comment in ecflow package, otherwise looks good

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Review by srherbener — APPROVED** [review]
Tested on my MacBook and it is working. Thanks for fixing the RPATH issue!

**Pull request merged**